src/sinks/socket.rs

```diff
@@ -221,37 +227,45 @@ mod test {
                 ..Default::default()
             },
         });
-        let stream = MaybeTlsSettings::from_config(&config, true).unwrap();
 
         // Only accept two connections.
-        let fut = stream
-            .bind(&addr)
-            .unwrap()
-            .incoming()
-            .take(2)
-            .for_each(move |mut socket| {
-                conn_counter1.fetch_add(1, Ordering::SeqCst);
-                loop {
-                    if let Ok(Async::Ready(_)) = close_rx.poll() {
-                        socket.get_ref().unwrap().shutdown(Shutdown::Write).unwrap();
-                    }
-
-                    let mut buf = vec![0u8; 11];
-                    match socket.read(&mut buf) {
-                        Err(error) if error.kind() == ErrorKind::WouldBlock => {}
-                        Err(error) => panic!("{}", error),
-                        Ok(n) => {
-                            if n == 0 {
-                                break;
-                            } else {
-                                msg_counter1.fetch_add(1, Ordering::SeqCst);
+        let jh = rt.spawn_handle_std(async move {
+            let tls = MaybeTlsSettings::from_config(&config, true).unwrap();
+            let mut listener = tls.bind(&addr).await.unwrap();
+            listener
+                .incoming()
+                .take(2)
+                .for_each(|connection| {
+                    let mut close_rx = close_rx.take();
```

Comment:
Why did this need to become an option?

Reply:
[tokio::sync::oneshot::Receiver](https://docs.rs/tokio/0.2.22/tokio/sync/oneshot/struct.Receiver.html) not `Copy` and not `Clone` and can be used only once, but we have two connections, so I need say for a second that I do not have `Receiver` to it.